### PR TITLE
Add to `resolve`'s public API

### PR DESCRIPTION
**Pull request by jseyfried**
This adds a public method `with_module_lexical_scope` to `Resolver` to allow users to resolve a path in the lexical scope of an arbitrary module.
r? @nrc

### CODE REVIEW

**Comment by jseyfried** [comment]
`rustdoc` needs to resolve in an arbitrary module's lexical scope for [RFC 1661](https://github.com/rust-lang/rfcs/pull/1661).
cc @JohnHeitmann

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -1413,29 +1413,36 @@ impl<'a> Resolver<'a> {
...
-                    };
+                    return None;
                 }
             }
         }
 
         None
     }
 
+    pub fn resolve_ident_in_module_lexical_scope(
```
Comment (nrc) [inline src/librustc_resolve/lib.rs]: Could you comment to explain what 'module lexical scope' means

File: src/librustc_resolve/lib.rs [hunk 2]
```diff
@@ -1413,29 +1413,36 @@ impl<'a> Resolver<'a> {
...
+        &mut self, module: Module<'a>, ident: ast::Ident, ns: Namespace, record_used: bool,
+    ) -> Option<NameBinding<'a>> {
```
Comment (nrc) [inline src/librustc_resolve/lib.rs]: nit: formatting

File: src/librustc_resolve/lib.rs (hunk 1 above)
Comment (jseyfried) [inline src/librustc_resolve/lib.rs]: Done

**Comment by jseyfried** [comment]
@nrc After thinking about this some more, I reimplemented this PR to provide a more powerful API.

**Comment by nrc** [comment]
iirc, `with_module_lexical_scope` is dead code now, whereas before it was used in the refactoring of an existing method. If that is not possible any more, then it would be good to add some sort of test for `with_module_lexical_scope` to make sure it is exercised somewhere.

**Comment by nrc** [comment]
otherwise, this lgtm

**Comment by jseyfried** [comment]
@nrc Hmm, we don't have a testing framework for `resolve` now (i.e. we don't have a way to get a `Resolver` in a testing environment).

There is no use for this function in the `resolve` today. Since it's only intended for use by `rustdoc` for RFC 1661, so I think it would be more appropriate to test it there.

However, don't have a way to get a `Resolver` in `rustdoc` either -- that will require refactoring `phase_2_configure_and_expand` to take a `ResolverArenas` and return a `Resolver` (a bit more involved). I could do that and add a test in a future PR or add it to this one.

**Comment by nrc** [comment]
OK, that sounds like a lot of work. Could you file an issue for a test so we don't forget, then add a `FIXME` with the issue number in the code? r=me with that

**Comment by jseyfried** [comment]
Sounds good, done. @bors r=nrc

**Pull request merged**